### DISCUSSION TITLE
Serve opensearch.xml with dynamic content

### DISCUSSION THREAD

**Discussion by andorfermichael** [nuxt/nuxt#9420] [Questions (Nuxt 2)] [answered]
I wondered what the best way was to serve opensearch.xml with dynamic content.

For example, I would like to call `https://example.com/opensearch` and to serve the following opensearch.xml file containing translations and template strings:

    <OpenSearchDescription
        xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
        <ShortName>$t('opensearch.title')</ShortName>
        <Description>$t('opensearch.description')</Description>
        <InputEncoding>UTF-8</InputEncoding>
        <Tags>$t('opensearch.tag')</Tags>
        <Image width="16" height="16" type="image/x-icon">{{ `https://example.${tld}/favicon.ico` }}</Image>
        <Url type="text/html" template="{{ `https://example.${tld}` }}">
            <Param name="query" value="{searchTerms}" />
        </Url>
        <moz:SearchForm>{{ `https://example.${tld}` }}</moz:SearchForm>
    </OpenSearchDescription>

**Comment by TheAlexLichter** [comment, marked as answer]
The way to go is possibly similar to the approach from the [feed-module](https://github.com/nuxt-community/feed-module) or [sitemap-module](https://github.com/nuxt-community/sitemap-module). You can use a serverMiddleware to render out the content you need and set the content dynamically.
